src/js/App/Sidenav/Navigation/DynamicNav.js

```diff
@@ -1,57 +1,60 @@
-import React, { useEffect } from 'react';
+import React, { useEffect, Fragment, useState } from 'react';
 import { useLoadModule } from '@scalprum/react-core';
 import PropTypes from 'prop-types';
 import { Skeleton, SkeletonSize } from '@redhat-cloud-services/frontend-components/Skeleton';
 import { NavItem } from '@patternfly/react-core';
 import { useHistory } from 'react-router';
 import { useDispatch, useSelector } from 'react-redux';
+import isEqual from 'lodash/isEqual';
+import ChromeNavItem from './ChromeNavItem';
 import { loadLeftNavSegment } from '../../../redux/actions';
 
-const DynamicNav = ({ dynamicNav }) => {
-  const {
-    location: { pathname },
-  } = useHistory();
-  const dispatch = useDispatch();
-  const [appName] = dynamicNav.split('/');
+const toArray = (value) => (Array.isArray(value) ? value : [value]);
+const mergeArrays = (orig, index, value) => [...orig.slice(0, index), ...toArray(value), ...orig.slice(index)];
+
+const HookedNavigation = ({ useNavigation, dynamicNav, pathname, ...props }) => {
   const currentNamespace = pathname.split('/')[1];
+  const [isLoaded, setIsLoaded] = useState(false);
+  const dispatch = useDispatch();
   const schema = useSelector(({ chrome: { navigation } }) => navigation[currentNamespace]);
-  const [{ default: navigation }] = useLoadModule({ appName, scope: appName, module: './Navigation' }, {});
+  const currNav = useSelector(({ chrome: { navigation } }) =>
+    navigation[currentNamespace]?.navItems?.filter((item) => item.dynamicNav === dynamicNav)
+  );
+  const newNav = useNavigation({ schema, dynamicNav, currentNamespace, currNav });
   useEffect(() => {
-    if (navigation) {
-      if (typeof navigation === 'function') {
-        const indexOfDynamicNav = schema.navItems.findIndex((nav) => nav.dynamicNav === dynamicNav);
-        if (indexOfDynamicNav !== -1) {
-          const { dynamicNav: _dynamicNav, ...originalNav } = schema.navItems[indexOfDynamicNav];
-          Promise.resolve(navigation({ schema, dynamicNav, currentNamespace })).then((data) => {
-            const newValue = Array.isArray(data)
-              ? data.map((item) => ({
-                  appId: dynamicNav.split('/')[0],
-                  ...originalNav,
-                  ...item,
-                }))
-              : [
-                  {
-                    ...originalNav,
-                    ...data,
-                  },
-                ];
-            dispatch(
-              loadLeftNavSegment(
-                {
-                  ...schema,
-                  navItems: schema.navItems.flatMap((item, key) => (key === indexOfDynamicNav ? newValue : item)),
-                },
-                currentNamespace,
-                pathname,
-                true
-              )
-            );
-          });
+    if (newNav?.length > 0) {
```

Comment:
So Navigation can still return an object... why do we only want arrays here?

Reply:
Ah, yeah! Good point. It should check just for the `newNav` not if it's an array.